test/spmd/tensor/test_pointwise_ops.py

```diff
@@ -90,6 +90,37 @@ def test_dropout_errors(self):
                 torch.nn.functional.dropout,
             )
 
+    @with_comms
+    def test_mul(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(self.rank)
+        spec = [Shard(0)]
+        input_tensor = torch.randn(
+            (8, 4), device=self.device_type, requires_grad=True
+        )
+        dtensor = DTensor.from_local(input_tensor, device_mesh, spec)
+        # Mutiplication with a scalar.
+        dt = torch.mul(dtensor, 8.0)
+        expected = torch.mul(input_tensor, 8.0)
+        self.assertEqual(input_tensor, dtensor.to_local())
+        self.assertEqual(expected, dt.to_local())
+
+        # Mutiplication with a matrix.
```

Comment:
can we group the "Mutiplication with a scalar" and "Mutiplication with a matrix" together?
i.e. 
```
for rhs in [8.0, other_dtensor]:
    res = torch.mul(dtensor, rhs)
    assertEqual(...
```

Reply:
I removed the scalar test since we are already testing torch.mul and that should be covered.